be/src/olap/rowset/segment_v2/segment.cpp

```diff
@@ -157,6 +156,19 @@ Status Segment::new_iterator(SchemaSPtr schema, const StorageReadOptions& read_o
         iter->reset(new SegmentIterator(this->shared_from_this(), schema));
     }
 
+    auto pruned_predicates = read_options.column_predicates;
+    auto pruned = false;
+    for (auto& it : _column_readers) {
+        if (it.second->prune_predicates_by_zone_map(pruned_predicates, it.first)) {
+            pruned = true;
+        }
+    }
+
+    if (pruned) {
```

Comment:
add a config to disable this
and the prune should only be used in query, not in compaction.